src/EnergyPlus/PCMThermalStorage.cc

```diff
@@ -0,0 +1,533 @@
+                            Units::C,
+                            state.dataLoopNodes->Node(PCM.UseSideInletNode).Temp,
+                            TimeStepType::System,
+                            StoreType::Average,
+                            PCM.Name);
+
+        SetupOutputVariable(state,
+                            "Thermal Energy Storage Use Side Outlet Temperature",
+                            Units::C,
+                            state.dataLoopNodes->Node(PCM.UseSideOutletNode).Temp,
+                            TimeStepType::System,
+                            StoreType::Average,
+                            PCM.Name);
+
+        SetupOutputVariable(state,
+                            "Thermal Energy Storage Use Side Mass Flow Rate",
+                            Units::kg_s,
+                            state.dataLoopNodes->Node(PCM.UseSideInletNode).MassFlowRate,
+                            TimeStepType::System,
+                            StoreType::Average,
+                            PCM.Name);
+
+        SetupOutputVariable(state,
+                            "Thermal Energy Storage Plant Side Inlet Temperature",
+                            Units::C,
+                            state.dataLoopNodes->Node(PCM.PlantSideInletNode).Temp,
+                            TimeStepType::System,
+                            StoreType::Average,
+                            PCM.Name);
+
+        SetupOutputVariable(state,
+                            "Thermal Energy Storage Plant Side Outlet Temperature",
+                            Units::C,
+                            state.dataLoopNodes->Node(PCM.PlantSideOutletNode).Temp,
+                            TimeStepType::System,
+                            StoreType::Average,
+                            PCM.Name);
+
+        SetupOutputVariable(state,
+                            "Thermal Energy Storage Plant Side Mass Flow Rate",
+                            Units::kg_s,
+                            state.dataLoopNodes->Node(PCM.PlantSideInletNode).MassFlowRate,
+                            TimeStepType::System,
+                            StoreType::Average,
+                            PCM.Name);
+    }
+
+    EnergyPlus::PCMStorage::PCMStorageData &EnergyPlus::PCMStorage::PCMStorageData::instance()
+    {
+        static EnergyPlus::PCMStorage::PCMStorageData inst;
+        return inst;
+    }
+
+    void GetPCMStorageInput(EnergyPlusData &state)
+    {
+        constexpr std::string_view RoutineName = "GetPCMStorageInput";
+
+        auto &PCM = EnergyPlus::PCMStorage::PCMStorageData::instance();
+        bool ErrorsFound = false;
+
+        int NumPCMObjs = state.dataInputProcessing->inputProcessor->getNumObjectsFound(state, "ThermalStorage:PCM");
+        if (NumPCMObjs != 1) {
+            ShowSevereError(state, "Exactly one ThermalStorage:PCM object is required.");
+            ErrorsFound = true;
+            return;
+        }
```

Comment:
Is it typical that the same TS material is used throughout the building? We could always change this later to allow more than 1 object but it seems a little restrictive.